app/poros/stats.rb

```diff
@@ -3,17 +3,123 @@ class Stats
               :saving_throws,
               :skills,
               :passive_perception,
-              :hit_points
+              :hit_points,
+              :proficiency_bonus,
+              :dex_attack,
+              :str_attack
 
-  def initialize(ancestry, sub_ancestry, background, class_data, score_type, traits)
-    # The line below will have to be updated once more subraces are created
-    ancestry[:asi] << sub_ancestry[:asi][0] if sub_ancestry != 'No Sub Ancestry'
-    @core_stats = CoreStats.new(ancestry[:asi], score_type)
+  def initialize(ancestry, sub_ancestry, background, class_data, score_type, traits, level)
+    @proficiency_bonus = 2
+    leveled_stats = level_up(level, class_data[:name].downcase) if level > 3
+    ancestry[:asi] << sub_ancestry[:asi].sample if sub_ancestry != 'No Sub Ancestry'
+    @core_stats = CoreStats.new(ancestry[:asi], score_type, leveled_stats)
     @saving_throws = SavingThrows.new(@core_stats.stats[:modifiers], class_data[:prof_saving_throws].split(', '))
     @skills = Skills.new(@core_stats.stats[:modifiers],
                          [background.skill_proficiency_one[0..-7].downcase,
-                          background.skill_proficiency_two[0..-7].downcase], class_data[:prof_skills].downcase, traits)
+                          background.skill_proficiency_two[0..-7].downcase], class_data[:prof_skills].downcase, traits, @proficiency_bonus)
     @passive_perception = @skills.skills[:perception] + 10
     @hit_points = @core_stats.stats[:modifiers][:con_mod] + class_data[:hp_at_1st_level].scan(/\d+/)[0].to_i
+    @dex_attack    = @core_stats.stats[:modifiers][:dex_mod] + @proficiency_bonus
+    @str_attack    = @core_stats.stats[:modifiers][:str_mod] + @proficiency_bonus
+    if level > 1
+      level.times do
+        @hit_points += @core_stats.stats[:modifiers][:con_mod] + class_data[:hp_at_higher_levels].scan(/\d+/)[2].to_i
+      end
+    end
+  end
+
+  def level_up(level, class_name)
+    if class_name == 'fighter'
+      level_up_fighter(level)
+    else
+      level_up_non_fighter(level)
+    end
+  end
```

Comment:
Ooooo boy, here's where the trouble starts.  Fighter's ability scores go up more often than anyone else's, so what lies below is a lot, and will need to be refactored.